---
title: "Waterwater Communities Alpha Diversity Revisited"
author: "Nate Olson"
date: "`r Sys.Date()`"
output:
  pdf_document: default
  html_notebook: default
---

## Objective
Re-evaluate $\alpha$-diversity analysis included in communities manuscript using more robust/ statistically sound methods to verify conclusions. 

## Approach  
* use breakaway to estimate richness  - using breakaway package  
* Diversity comparisons  
    * Influent across sites  
    * Influent-Effluent pairs  
    * Across treatment process by site  
    * WWTP to SI site  

## Summary
Large range in richness estimates for all samples. 
Because breakaway only provides estimates, standard errors, upper, and lower limits, we cannot use the same statistical test that we used prevously. 
The breakaway package has a statistical test that I don't fully understand the statistical test and some of the resutls are not consistent with my expectation. 
For example the influent estimate for MWTP1 is only consistent with one of the three samples. 
Additionally, the pairs plots cannot be used when we have range and point estimates. 

```{r warning=FALSE, message=FALSE, echo = FALSE}
source("joe_diversity_functions.R")
library(tidyverse)
library(stringr)
library(forcats)
library(metagenomeSeq)
library(vegan)
library(vegetarian)
library(iNEXT)
library(breakaway)
```

## Load and Tidy data 
```{r warning=FALSE, message=FALSE}
project_dir <- file.path("/Users/nolson/Projects/sapkota_wastewater/wastewater16S/")
metadata_dir <- file.path(project_dir, "metadata/")

dat_files <- c(
   env_meta = file.path(metadata_dir, "env_meta.csv"),
   seq_meta = file.path(metadata_dir, "seq_meta.csv"),
   id_key = file.path(metadata_dir, "id_keys.csv"),
   merge_id = file.path(metadata_dir, "merge_id_key.csv"),
   key_meta = file.path(metadata_dir, "key_meta.csv"),
   mr_blast_file = file.path(project_dir, "wastewater_blast.rdata")
)

# metadata
meta_env_src <- read_csv(dat_files['env_meta'])


# Loading blast based OTU count data.
load(dat_files['mr_blast_file'])

# Only including spray irrigation and treatment plant data in analysis.
plantObj <- which((pData(wastewaterMRobj_blast)$study_type == "plant" &
                       pData(wastewaterMRobj_blast)$Stage_name != "Influent post screening") |
                      pData(wastewaterMRobj_blast)$Region_Site == "MA_SI1") %>% 
                      {wastewaterMRobj_blast[,.]} # %>% cumNorm(p = 0.75)

ww_meta <- plantObj %>% pData() %>% 
      as.data.frame %>% mutate(Seq_ID = row.names(.)) %>% 
      separate(Env_ID,c("Region","TP","Date","Stage"), sep = "_") %>% 
      unite("WW_TP", Region,TP, sep = "_", remove = F)  %>%
      mutate(plant_name = paste(ifelse(Region == "MA", 
                                       "Mid-Atlantic",
                                       "Midwest"),
                                ifelse(TP %in% c("TP1","TP2"),
                                       paste0("WW",TP),TP))) %>% 
  mutate(Stage = if_else(Stage == "RawInfluent", "Influent", Stage)) %>%
  mutate(Stage = if_else(Stage == "AfterUVTreatment", "After UV", Stage)) %>%
  mutate(Stage = if_else(Stage == "BeforeUVTreatment", "Before UV", Stage)) %>%
  mutate(Stage = if_else(Stage == "HoldingPondInlet", "Pond", Stage)) %>%
  mutate(Stage = if_else(Stage == "PumpHouseInlet", "Inlet to Pumphouse", Stage)) %>%
  mutate(Stage = if_else(Stage == "ActivatedSludgeReactor", "Activated Sludge Reactor", Stage)) %>%
  mutate(Stage = if_else(Stage == "SecondaryClarifier", "Secondary Clarifier", Stage)) %>%
  mutate(Stage = if_else(Stage == "CellB", "Cell B", Stage)) %>%
  mutate(Stage = if_else(Stage == "PostAeration", "Post Aeration", Stage)) %>%
      select(Seq_ID, WW_TP, plant_name, Region, TP, Date, Stage, Stage_name, 
             study_type, Season, StageID, StageID_all_plants)

rownames(ww_meta) <- ww_meta$Seq_ID
      
pData(plantObj) <- ww_meta 

## filtering samples with less than 100 reads
plantObj <- which(colSums(plantObj) > 100) %>% {plantObj[, .]} 

## Count Matrix for diversity estimates
count_mat <- plantObj@assayData$counts
sam_dat <- pData(plantObj) %>% rownames_to_column(var = "samID")
```

## Breakaway Richness Estimates
Breakaway using a statisticall modeling approach to infer the number of unobserved OTUs (OTUs with counts of 0). 
Predicted values are not observed and open circles are not defined in legend. 

```{r}
# count_vec <- count_list[[2]] 
# count_tbl <- count_vec[count_vec > 1] %>% table() %>% as.data.frame() 
# colnames(count_tbl) <- c("freq","count")
# 
# breakaway_nof1(count_tbl)
```
Check warning message....

```{r}
freq_df <- count_mat %>% 
    as_data_frame() %>% 
    rownames_to_column(var = "otuID")  %>% 
    gather("samID","abundance",-otuID) %>% 
    group_by(samID, abundance) %>% 
    summarise(freq = n()) %>% 
    filter(abundance > 1) %>% 
    group_by(samID) %>% 
    nest() 

break_fit <- freq_df$data %>% 
    map(as.data.frame) %>%
    map(breakaway_nof1, answer = TRUE,
        plot = FALSE, force = TRUE)
```
Generate a data frame with breakaway richness estimates (est), standard error (seest), and confidence interval (lower, upper).
```{r}
## is_numeric is deprecated
break_est <- break_fit %>% set_names(freq_df$samID)  %>% map(~.$est) %>%
  keep(is.numeric) %>% as_tibble() %>% gather("samID","est")
break_seest <- break_fit %>% set_names(freq_df$samID)  %>% map(~.$seest) %>%
  keep(is.numeric) %>% as_tibble() %>% gather("samID","seest")
break_ci <- break_fit %>% set_names(freq_df$samID)  %>% map(~.$ci) %>%
  keep(is.numeric) %>% as_tibble() %>% mutate(ci = c("lower","upper")) %>%
  gather("samID","value",-ci) %>% spread(ci, value)
break_model <- break_fit %>% set_names(freq_df$samID)  %>% map(~.$name) %>% 
    keep(is.character) %>% as_tibble() %>% gather("samID","model")
break_df <- left_join(break_est, break_seest) %>% 
    left_join(break_ci) %>% left_join(break_model) %>% 
    rename(Seq_ID = samID) %>% 
    left_join(ww_meta) %>% 
    rename(samID = Seq_ID)
```

Breakdown of the number of samples each model was used for the richness estimate. 
```{r}
break_df %>% ggplot() + geom_bar(aes(x = model))
```


Breakaway richness failed for two samples.
```{r}
break_fit %>% set_names(freq_df$samID)  %>% map(~.$est) %>%
  keep(~!is.numeric(.))
```


### Influent Across Sites

```{r}
inf_df <- filter(break_df, Stage == "Influent") %>% 
    mutate(plant_name = factor(plant_name)) 
    
           
plant_fac <- inf_df$plant_name

design_mat <- model.matrix(~ plant_fac + 0)
colnames(design_mat) <- colnames(design_mat) %>% str_replace("plant_fac","")
inf_betta <- betta(chats = inf_df$est, ses = inf_df$seest, X = design_mat)
```

__Take these results with a grain of salt.__ The math used in the statistical test is beyond me. 

Results from the betta statistical test. Results indicate that the MATP1 estimate is significantly different from zero and that MATP2 and MWTP1 but MWTP2 are significantly different from MATP1. 
The one sample MWTP1 sample with the lower estimate is driving the MWTP1 estimate, and therefore a statistical artifact. 
When the one sample is excluded MWTP1 is no longer significantly different from WWTP1. 

```{r}
inf_betta$table
```

```{r}
inf_df_v2 <- filter(break_df, Stage == "Influent", 
                 est > 200, plant_name != "Midwest WWTP2") %>% 
    mutate(plant_name = factor(plant_name)) 
    
           
plant_fac <- inf_df_v2$plant_name

design_mat <- model.matrix(~ plant_fac + 0)
colnames(design_mat) <- colnames(design_mat) %>% str_replace("plant_fac","")
inf_betta_v2 <- betta(chats = inf_df_v2$est, ses = inf_df_v2$seest, X = design_mat)
```

```{r}
inf_betta_v2$table
```

Plot below shows the point estimates and ranges (upper and lower estimate) for influent sample by treatment plant. Black point and error bars indicate the richness estiamte +/- standard errors for individual treatment plants calculated using the `betta` function in breakaway. 
```{r}
inf_tbl <- inf_betta$table %>% 
    as.data.frame() %>% 
    rownames_to_column(var = "plant_name") %>% 
    mutate(ymin = Estimates - `Standard Errors`,
           ymax = Estimates + `Standard Errors`,
           ymin = if_else(ymin < 0, 1, ymin))

inf_df %>% 
    group_by(plant_name) %>% 
    mutate( sam_num = 1:n(), mid_sam = median(sam_num), 
            x_pos = as.numeric(plant_name),
            x_pos = if_else(sam_num >= mid_sam, 
                            x_pos + 0.05 * (sam_num - floor(mid_sam)),  
                            x_pos - 0.05 * sam_num)) %>% 
    ggplot(aes(x = plant_name)) + 
    geom_blank() + 
    geom_segment(aes(x = x_pos, xend = x_pos, 
                     y = lower, yend = upper, 
                     color = plant_name), lwd = 1) +
    geom_point(aes(x = x_pos, y = est, color = plant_name)) + 
    geom_point(data = inf_tbl, aes(x = plant_name, y = Estimates)) + 
    geom_errorbar(data = inf_tbl, aes(x = plant_name, ymax = ymax, ymin = ymin),
                  width = 0.1) + 
    scale_y_log10() + 
    annotation_logticks(sides = "l") + 
    theme_bw() + 
    theme(legend.position = "none") + 
    labs(x = "Treatment Plant", y = "Richness Estimate")
```

Same plot without plant estimates - this is the one I prefer
```{r}
inf_df %>% 
    group_by(plant_name) %>% 
    mutate( sam_num = 1:n(), mid_sam = median(sam_num), 
            x_pos = as.numeric(plant_name),
            x_pos = if_else(sam_num >= mid_sam, 
                            x_pos + 0.05 * (sam_num - floor(mid_sam)),  
                            x_pos - 0.05 * sam_num)) %>% 
    ggplot(aes(x = plant_name)) + 
    geom_blank() + 
    geom_segment(aes(x = x_pos, xend = x_pos, 
                     y = lower, yend = upper, 
                     color = plant_name), lwd = 1) +
    geom_point(aes(x = x_pos, y = est, fill = plant_name), shape = 21) + 
    labs(x = "Samples", y = "Richness Estimate") + 
    scale_y_log10() + 
    annotation_logticks(sides = "l") + 
    theme_bw() + 
    theme(legend.position = "none") + 
    labs(x = "Treatment Plant")
```





### Influent-Effluent Pairs 
Interpretation of statistical test results, Influent is significantly different from 0. 
Effluent is significantly different with a lower abundance from influent.

Sorry we lost the pair info. I can try to generate a new plot with pairs but pair information is not included in the statistical test.
```{r}
df <- filter(break_df, Stage %in% c("Influent","Effluent")) %>% 
    mutate(Stage = factor(Stage, level = c("Influent","Effluent")))
stage_fac <- df$Stage

design_mat <- model.matrix(~ stage_fac + 0)
colnames(design_mat) <- colnames(design_mat) %>% str_replace("stage_fac","")

df_betta <- betta(chats = df$est, ses = df$seest, X = design_mat)

df_tbl <- df_betta$table %>% 
    as.data.frame() %>% 
    rownames_to_column(var = "Stage") %>% 
    mutate(ymin = Estimates - `Standard Errors`,
           ymax = Estimates + `Standard Errors`,
           ymin = if_else(ymin < 0, 1, ymin))
df_tbl
```



```{r}
df %>% 
    group_by(Stage) %>% 
    mutate( sam_num = 1:n(), mid_sam = median(sam_num), 
            x_pos = as.numeric(Stage),
            x_pos = if_else(sam_num >= mid_sam, 
                            x_pos + 0.025 * (sam_num - floor(mid_sam)),  
                            x_pos - 0.025 * sam_num)) %>% 
    ggplot(aes(x = Stage)) + 
    geom_blank() + 
    geom_segment(aes(x = x_pos, xend = x_pos, 
                     y = lower, yend = upper, 
                     color = Stage), lwd = 1) +
    geom_point(aes(x = x_pos, y = est, color = Stage)) + 
    geom_point(data = df_tbl, aes(x = Stage, y = Estimates)) + 
    geom_errorbar(data = df_tbl, aes(x = Stage, ymax = ymax, ymin = ymin),
                  width = 0.1) + 
    scale_y_log10() + 
    annotation_logticks(sides = "l") + 
    theme_bw() + 
    theme(legend.position = "none") + 
    labs(x = "Stage", y = "Richness Estimate")
```





## Treatment Process
### MA_TP1
```{r}
df <- break_df %>% 
    filter(study_type == "plant", WW_TP == "MA_TP1") %>% 
    # mutate(Stage = factor(Stage,levels = c("Influent","Activated Sludge Reactor", 
    #                                        "Post Aeration","Secondary Clarifier",
    #                                        "CellB","Effluent"))) 
    ## Excluding Post Aeration and CellB as richness estimates were not obtianed for these samples
    mutate(Stage = factor(Stage,levels = c("Influent","Activated Sludge Reactor","Secondary Clarifier","Effluent"))) 

stage_fac <- df$Stage

design_mat <- model.matrix(~ stage_fac + 0)

colnames(design_mat) <- colnames(design_mat) %>% str_replace("stage_fac","")

df_betta <- betta(chats = df$est, ses = df$seest, X = design_mat)

df_tbl <- df_betta$table %>% 
    as.data.frame() %>% 
    rownames_to_column(var = "Stage") %>% 
    mutate(ymin = Estimates - `Standard Errors`,
           ymax = Estimates + `Standard Errors`,
           ymin = if_else(ymin < 0, 1, ymin))
df_tbl
```



```{r}
df %>% 
    group_by(Stage) %>% 
    mutate( sam_num = 1:n(), mid_sam = median(sam_num), 
            x_pos = as.numeric(Stage),
            x_pos = if_else(sam_num >= mid_sam, 
                            x_pos + 0.05 * (sam_num - floor(mid_sam)),  
                            x_pos - 0.05 * sam_num)) %>% 
    ggplot(aes(x = Stage)) + 
    geom_blank() + 
    geom_segment(aes(x = x_pos, xend = x_pos, 
                     y = lower, yend = upper, 
                     color = Stage), lwd = 1) +
    geom_point(aes(x = x_pos, y = est, color = Stage)) + 
    geom_point(data = df_tbl, aes(x = Stage, y = Estimates)) + 
    geom_errorbar(data = df_tbl, aes(x = Stage, ymax = ymax, ymin = ymin),
                  width = 0.1) + 
    scale_y_log10() + 
    annotation_logticks(sides = "l") + 
    theme_bw() + 
    theme(legend.position = "none") + 
    labs(x = "Stage", y = "Richness Estimate")
```

### MA_TP2
```{r}
df <- break_df %>% 
    filter(study_type == "plant", WW_TP == "MA_TP2") %>% 
    # mutate(Stage = factor(Stage,levels = c("Influent","Activated Sludge Reactor",
    #                                        "Post Aeration","Secondary Clarifier",
    #                                        "CellB","Effluent")))
    ## Excluding Post Aeration and CellB as richness estimates were not obtianed for these samples
    mutate(Stage = factor(Stage,levels = c("Influent","Activated Sludge Reactor","Secondary Clarifier")))

stage_fac <- df$Stage

design_mat <- model.matrix(~ stage_fac + 0)

colnames(design_mat) <- colnames(design_mat) %>% str_replace("stage_fac","")

df_betta <- betta(chats = df$est, ses = df$seest, X = design_mat)

df_tbl <- df_betta$table %>% 
    as.data.frame() %>% 
    rownames_to_column(var = "Stage") %>% 
    mutate(ymin = Estimates - `Standard Errors`,
           ymax = Estimates + `Standard Errors`,
           ymin = if_else(ymin < 0, 1, ymin))
df_tbl
```

```{r}
df %>% 
    group_by(Stage) %>% 
    mutate( sam_num = 1:n(), mid_sam = median(sam_num), 
            x_pos = as.numeric(Stage),
            x_pos = if_else(sam_num >= mid_sam, 
                            x_pos + 0.05 * (sam_num - floor(mid_sam)),  
                            x_pos - 0.05 * sam_num)) %>% 
    ggplot(aes(x = Stage)) + 
    geom_blank() + 
    geom_segment(aes(x = x_pos, xend = x_pos, 
                     y = lower, yend = upper, 
                     color = Stage), lwd = 1) +
    geom_point(aes(x = x_pos, y = est, color = Stage)) + 
    geom_point(data = df_tbl, aes(x = Stage, y = Estimates)) + 
    geom_errorbar(data = df_tbl, aes(x = Stage, ymax = ymax, ymin = ymin),
                  width = 0.1) + 
    scale_y_log10() + 
    annotation_logticks(sides = "l") + 
    theme_bw() + 
    theme(legend.position = "none") + 
    labs(x = "Stage", y = "Richness Estimate")
```

### MW_TP1
```{r}
df <- break_df %>% 
    filter(study_type == "plant", WW_TP == "MW_TP1") %>% 
    # mutate(Stage = factor(Stage,levels = c("Influent","Activated Sludge Reactor",
    #                                        "Post Aeration","Secondary Clarifier",
    #                                        "CellB","Effluent")))
    ## Excluding Post Aeration and CellB as richness estimates were not obtianed for these samples
    mutate(Stage = factor(Stage, levels = c("Influent", "Post Aeration","Secondary Clarifier", "Effluent")))

stage_fac <- df$Stage

design_mat <- model.matrix(~ stage_fac + 0)

colnames(design_mat) <- colnames(design_mat) %>% str_replace("stage_fac","")

df_betta <- betta(chats = df$est, ses = df$seest, X = design_mat)

df_tbl <- df_betta$table %>% 
    as.data.frame() %>% 
    rownames_to_column(var = "Stage") %>% 
    mutate(ymin = Estimates - `Standard Errors`,
           ymax = Estimates + `Standard Errors`,
           ymin = if_else(ymin < 0, 1, ymin))
df_tbl
```

```{r}
df %>% 
    group_by(Stage) %>% 
    mutate( sam_num = 1:n(), mid_sam = median(sam_num), 
            x_pos = as.numeric(Stage),
            x_pos = if_else(sam_num >= mid_sam, 
                            x_pos + 0.05 * (sam_num - floor(mid_sam)),  
                            x_pos - 0.05 * sam_num)) %>% 
    ggplot(aes(x = Stage)) + 
    geom_blank() + 
    geom_segment(aes(x = x_pos, xend = x_pos, 
                     y = lower, yend = upper, 
                     color = Stage), lwd = 1) +
    geom_point(aes(x = x_pos, y = est, fill = Stage), shape = 21) + 
    # geom_point(aes(x = x_pos, y = est, color = Stage)) + 
    # geom_point(data = df_tbl, aes(x = Stage, y = Estimates)) + 
    # geom_errorbar(data = df_tbl, aes(x = Stage, ymax = ymax, ymin = ymin),
    #               width = 0.1) + 
    scale_y_log10() + 
    annotation_logticks(sides = "l") + 
    theme_bw() + 
    theme(legend.position = "none") + 
    labs(x = "Stage", y = "Richness Estimate")
```


### MW_TP2
```{r}
df <- break_df %>% 
    filter(study_type == "plant", WW_TP == "MW_TP2") %>% 
    # mutate(Stage = factor(Stage,levels = c("Influent","Activated Sludge Reactor",
    #                                        "Post Aeration","Secondary Clarifier",
    #                                        "CellB","Effluent")))
    ## Only including stages with samples with richness estimtates
    mutate(Stage = factor(Stage,levels = c("Influent", "Cell B","Effluent")))

stage_fac <- df$Stage

## Unable to perform test - too few samples, I think ....
design_mat <- model.matrix(~ stage_fac + 0)

colnames(design_mat) <- colnames(design_mat) %>% str_replace("stage_fac","")

df_betta <- betta(chats = df$est, ses = df$seest, X = design_mat)

df_tbl <- df_betta$table %>%
    as.data.frame() %>%
    rownames_to_column(var = "Stage") %>%
    mutate(ymin = Estimates - `Standard Errors`,
           ymax = Estimates + `Standard Errors`,
           ymin = if_else(ymin < 0, 1, ymin))
df_tbl
```


```{r}
df %>% 
    group_by(Stage) %>% 
    mutate( sam_num = 1:n(), mid_sam = median(sam_num), 
            x_pos = as.numeric(Stage),
            x_pos = if_else(sam_num >= mid_sam, 
                            x_pos + 0.05 * (sam_num - floor(mid_sam)),  
                            x_pos - 0.05 * sam_num)) %>% 
    ggplot(aes(x = Stage)) + 
    geom_blank() + 
    geom_segment(aes(x = x_pos, xend = x_pos, 
                     y = lower, yend = upper, 
                     color = Stage), lwd = 1) +
    geom_point(aes(x = x_pos, y = est, fill = Stage), shape = 21) + 
    # geom_point(aes(x = x_pos, y = est, color = Stage)) + 
    # geom_point(data = df_tbl, aes(x = Stage, y = Estimates)) +
    # geom_errorbar(data = df_tbl, aes(x = Stage, ymax = ymax, ymin = ymin),
    #               width = 0.1) +
    scale_y_log10() + 
    annotation_logticks(sides = "l") + 
    theme_bw() + 
    theme(legend.position = "none") + 
    labs(x = "Stage", y = "Richness Estimate")
```

### Influent to Spray

```{r}
pd <- position_dodge(width = 0.1)

break_df %>% filter( WW_TP %in% c("MA_TP1","MA_SI1")) %>% 
    filter(Stage %in% c("Influent","Effluent", "Before UV", 
                        "After UV","Pond", 
                        "Inlet to Pumphouse")) %>% 
    mutate(Stage = factor(Stage,levels = c("Influent","Effluent","Before UV", 
                                           "After UV","Pond", 
                                           "Inlet to Pumphouse"))) %>% 
    ggplot(aes(x = samID, y = est,  color = Date)) +
    # geom_line(aes(group = Date), color = "grey60") +
    geom_point() +
    geom_errorbar(aes(ymin = lower, ymax = upper), width = 0) +
labs(y = "Diversity Metric", color = "WWTP") + theme_bw() +
    theme(axis.text.x = element_blank()) + 
    scale_y_log10() + 
    facet_wrap(~Stage, scales = "free_x", nrow = 1)
```

```{r}
df <- break_df %>% 
    filter( WW_TP %in% c("MA_TP1","MA_SI1")) %>% 
    filter(Stage %in% c("Influent","Effluent", "Before UV", 
                        "After UV","Pond", 
                        "Inlet to Pumphouse")) %>% 
    mutate(Stage = factor(Stage,levels = c("Influent","Effluent","Before UV", 
                                           "After UV","Pond", 
                                           "Inlet to Pumphouse")))

stage_fac <- df$Stage

## Unable to perform test - too few samples, I think ....
design_mat <- model.matrix(~ stage_fac + 0)

colnames(design_mat) <- colnames(design_mat) %>% str_replace("stage_fac","")

df_betta <- betta(chats = df$est, ses = df$seest, X = design_mat)

df_tbl <- df_betta$table %>%
    as.data.frame() %>%
    rownames_to_column(var = "Stage") %>%
    mutate(ymin = Estimates - `Standard Errors`,
           ymax = Estimates + `Standard Errors`,
           ymin = if_else(ymin < 0, 1, ymin))
df_tbl
```


```{r}
df %>% 
    group_by(Stage) %>% 
    mutate( sam_num = 1:n(), mid_sam = median(sam_num), 
            x_pos = as.numeric(Stage),
            x_pos = if_else(sam_num >= mid_sam, 
                            x_pos + 0.05 * (sam_num - floor(mid_sam)),  
                            x_pos - 0.05 * sam_num)) %>% 
    ggplot(aes(x = Stage)) + 
    geom_blank() + 
    geom_segment(aes(x = x_pos, xend = x_pos, 
                     y = lower, yend = upper, 
                     color = Stage), lwd = 1) +
    geom_point(aes(x = x_pos, y = est, color = Stage)) + 
    geom_point(data = df_tbl, aes(x = Stage, y = Estimates)) +
    geom_errorbar(data = df_tbl, aes(x = Stage, ymax = ymax, ymin = ymin),
                  width = 0.1) +
    scale_y_log10() + 
    annotation_logticks(sides = "l") + 
    theme_bw() + 
    theme(legend.position = "none") + 
    labs(x = "Stage", y = "Richness Estimate")
```


Use for shannon with breakaway
```{r}
# freq_df %>% left_join(break_df) %>% filter(!is.na(est))

## Need to provide est and ssest estimates from breakaway no 1
# shannon_est <- freq_df$data %>% 
#     map(as.data.frame) %>%
#     map(shannon_better)
```

